Added documentation for Presto on Spark

Ravion opened the pull request:
Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
* ...
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

arhimondr commented:
@Ravion going to review it today or tomorrow. Sorry for the delay.

arhimondr commented on presto-docs/src/main/sphinx/installation/spark.rst:
```diff
@@ -0,0 +1,44 @@
+
+.. code-block:: none
+
+    query.hash-partition-count=10
```
Actually the default `config.properties` file can be empty.

arhimondr commented on presto-docs/src/main/sphinx/installation/spark.rst:
```diff
@@ -0,0 +1,44 @@
+    presto-spark-launcher-\ |version|\ .jar \
+    --package presto-spark-package-\ |version|\ .tar.gz \
+    --config /presto/etc/config.properties \
+    --catalogs /presto/etc/catalogs \
```
Could you please add a paragrapth describing this catalog structure (or point it to the section from the Presto documentation https://prestodb.io/docs/current/installation/deployment.html#catalog-properties

arhimondr commented on presto-docs/src/main/sphinx/installation/spark.rst:
```diff
@@ -0,0 +1,44 @@
+
+    /spark/bin/spark-submit \
+    --master spark://spark-master:7077 \
+    --executor-cores 4 \
```
Please add a paragraph explaining this. We need to set `spark.task.cpus` equal to the `--executor-cores` to make sure only a single Presto on Spark task is run in a single Spark executor. We introduced this limitation to avoid duplicating broadcasted hash table for every task. We may optimize this in the future and remove this limitation. Also the `task.concurrency`, `task.max-worker-threads` (and possibly `task.writer-count`) should be adjusted to keep all executor cores busy. For example if `--executor-cores 4` the `task.concurrency`, `task.max-worker-threads` must be set to `4`. You can actually include this properties in the `config.properties` example.

Ravion commented:
@arhimondr Updated the changes..thanks for guidance. Please review

arhimondr commented:
@Ravion It looks like the rebase didn't go well. Could you please try to rebase again to drop all unrelated changes?

Ravion commented:
@Andrill..My sincere apologies. It got messed up. Kindly close this PR. I have updated everything in Added documentation for Presto on Spark #14768 , a new PR and added all changes
